app/src/main/java/edu/uc/mille5bm/smartnutrition/BMIActivity.kt

```diff
@@ -1,101 +1,95 @@
-
 package edu.uc.mille5bm.smartnutrition
 
 import android.graphics.Color
 import android.os.Bundle
-import androidx.appcompat.app.AppCompatActivity
 import android.view.View
-import android.widget.*
-import androidx.cardview.widget.CardView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import edu.uc.mille5bm.smartnutrition.databinding.ActivityBmiCalculatorBinding
 import edu.uc.mille5bm.smartnutrition.ui.main.BMICalcUtil
 import edu.uc.mille5bm.smartnutrition.ui.main.BMICalcUtil.Companion.BMI_CATEGORY_HEALTHY
 import edu.uc.mille5bm.smartnutrition.ui.main.BMICalcUtil.Companion.BMI_CATEGORY_OBESE
 import edu.uc.mille5bm.smartnutrition.ui.main.BMICalcUtil.Companion.BMI_CATEGORY_OVERWEIGHT
 import edu.uc.mille5bm.smartnutrition.ui.main.BMICalcUtil.Companion.BMI_CATEGORY_UNDERWEIGHT
 
 
-class BMIActivity: AppCompatActivity(){
-    lateinit var weightKgEditText: EditText
-    lateinit var heightCmEditText: EditText
-    lateinit var weightLbsEditText: EditText
-    lateinit var heightFtEditText: EditText
-    lateinit var heightInEditText: EditText
-    lateinit var calculateButton: Button
-    lateinit var bmiTextView: TextView
-    lateinit var categoryTextView: TextView
-    lateinit var toggleUnitsButton: ToggleButton
-    lateinit var bmiResultCardView: CardView
+class BMIActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityBmiCalculatorBinding
     var inMetricUnits = false
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_bmi_calculator)
-        weightKgEditText = findViewById(R.id.activity_main_weightkgs)
-        heightCmEditText = findViewById(R.id.activity_main_heightcm)
-        weightLbsEditText = findViewById(R.id.activity_main_weightlbs)
-        heightFtEditText = findViewById(R.id.activity_main_heightfeet)
-        heightInEditText = findViewById(R.id.activity_main_heightinches)
-        calculateButton = findViewById(R.id.activity_main_calculate)
-        toggleUnitsButton = findViewById(R.id.activity_main_toggleunits)
-        bmiTextView = findViewById(R.id.activity_main_bmi)
-        categoryTextView = findViewById(R.id.activity_main_category)
-        bmiResultCardView = findViewById(R.id.activity_main_resultcard)
+        binding = ActivityBmiCalculatorBinding.inflate(layoutInflater)
+        setContentView(binding.root)
```

Comment:
General implementation for view binding in an activity

Reply:
Ah, neat.  This saves a lot of code.